mods/default/functions.lua

```diff
@@ -481,3 +481,35 @@ minetest.register_abm({
 		end
 	end
 })
+
+
+--
+-- Checks if specified volume intersects a protected volume
+--
+
+-- Checks 7 corners first then interior volume at specified intervals
+-- minp can be skipped as it is checked in the loop below
+
+function default.intersects_protection(minp, maxp, name, interval)
+	if minetest.is_protected({x = minp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = maxp.z}, name) then
+		return true
+	end
+
+	for z = minp.z, maxp.z, interval do
+	for y = minp.y, maxp.y, interval do
+	for x = minp.x, maxp.x, interval do
```

Comment:
The corners are usually not rechecked because interval is rarely a perfect division of volume dimensions.